pkg/server/conn.go

```diff
@@ -358,17 +358,20 @@ func (cc *clientConn) handshake(ctx context.Context) error {
 func (cc *clientConn) Close() error {
 	cc.server.rwlock.Lock()
 	delete(cc.server.clients, cc.connectionID)
-	connections := len(cc.server.clients)
+	delete(cc.server.resourceGroupMap, cc.getCtx().GetSessionVars().ResourceGroupName)
 	cc.server.rwlock.Unlock()
-	return closeConn(cc, connections)
+	return closeConn(cc, cc.server.resourceGroupMap)
 }
 
 // closeConn is idempotent and thread-safe.
 // It will be called on the same `clientConn` more than once to avoid connection leak.
-func closeConn(cc *clientConn, connections int) error {
+func closeConn(cc *clientConn, connections map[string]int) error {
 	var err error
 	cc.closeOnce.Do(func() {
-		metrics.ConnGauge.Set(float64(connections))
+		for name, count := range connections {
```

Comment:
Why not directly do `metrics.ConnGauge.WithLabelValues(name).Dec()`

Reply:
some caller may close many connections.